### PR TITLE
:tada: Add leaflet-vue-amplify

### PR DESCRIPTION
**Pull request by dayjournal**
Hi,
This example demonstrates how to use Leaflet with Amazon Location Service and AWS Amplify within a Vue.js application.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hello, thanks again for contributing to this repo. I will review the PR this week and provide feedback.

File: leaflet-vue-amplify/src/components/MapPane.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    </div>
+</template>
+
+<script>
+import L from 'leaflet';
+import '@maplibre/maplibre-gl-leaflet';
+import { Auth, Signer } from 'aws-amplify';
+import awsconfig from '../aws-exports';
+
+export default {
```
Comment (dreamorosi) [inline leaflet-vue-amplify/src/components/MapPane.vue:13]: The `transformRequest` method could be removed by taking a dependency on `maplibre-gl-js-amplify` a package mentioned [in the Amplify Geo docs](https://docs.amplify.aws/lib/geo/maps/q/platform/js/). This package, among other things, exports a class called `AmplifyMapLibreRequest` which basically implements a similar function to this one you're using.

Additionally I would recommend getting the name/style of the map at line 27 using the `Geo` module like so: `Geo.getDefaultMap().mapName`.

Finally, I would avoid hardcoding the attribution as other Customers might be using different map styles (i.e. Esri ones). To make it dynamic we could get the attribution from the map's `style-descriptor` (the same retrieved at line 44 of your implementation). This file is a JSON with a lot of info about the map, including the attribution.

The new implementation would look like this (I've tested it while reviewing both with Esri & HERE maps):
```js
import { Auth } from "aws-amplify";
import { Geo } from "@aws-amplify/geo";
import { AmplifyMapLibreRequest } from "maplibre-gl-js-amplify";

export default {
  name: "MapPane",
  mounted: async function () {
    const credentials = await Auth.currentCredentials();
    const { transformRequest } = new AmplifyMapLibreRequest(
      credentials,
      Geo.getDefaultMap().region
    );
    const attribution = await this.getAttribution(transformRequest);
    this.mapCreate(attribution, transformRequest);
  },
  methods: {
    getAttribution: async (transformRequest) => {
      try {
        const res = await fetch(
          transformRequest(
            `https://maps.geo.${
              Geo.getDefaultMap().region
            }.amazonaws.com/maps/v0/maps/${
              Geo.getDefaultMap().mapName
            }/style-descriptor`
          ).url
        );
        const data = await res.json();
        if (Geo.getDefaultMap().style.indexOf("Esri") !== -1) {
          return data.sources.esri.attribution;
        } else if (Geo.getDefaultMap().style.indexOf("Here") !== -1) {
          return data.sources.omv.attribution;
        }
      } catch (err) {
        console.error(err);
      }
    },
    mapCreate: async (attribution, requestTransformer) => {
      const sample = L.maplibreGL({
        style: Geo.getDefaultMap().mapName,
        attribution: attribution,
        transformRequest: requestTransformer,
      });
      const map = L.map("map", {
        center: [35.681, 139.767],
        zoom: 14,
        zoomControl: true,
        layers: [sample],
      });
      const Map_BaseLayer = {
        sample: sample,
      };
      L.control.layers(Map_BaseLayer, null).addTo(map);
    },
  },
};
```

**Comment by dayjournal** [comment]
@dreamorosi 
Hi, Thanks for the suggestion to use maplibre-gl-js-amplify and amplify geo.
I have checked and fixed the code. I also checked that it works.

スクリーンショット 2021-12-26 21 44 28

**Review by dreamorosi — APPROVED** [review]
Looks good to me, thanks again for contributing!

Happy holidays🎄!!

**Comment by dayjournal** [comment]
Thank you!
Happy holidays!

**Pull request merged**